src/fpylll/gso.pyx

```diff
@@ -13,6 +13,11 @@ from util cimport preprocess_indices, check_float_type
 from fpylll cimport mpz_double, mpz_ld, mpz_dpe, mpz_mpfr, mpz_dd, mpz_qd
 from fplll cimport FT_DOUBLE, FT_LONG_DOUBLE, FT_DD, FT_QD, FT_DPE, FT_MPFR, FloatType
 
+import cython
+import numpy as np
```

Comment:
Well, we'd be looping in Cython :) I presume you need the `double*` because it is the interface for another C/C++ program, then? 

Mhhh, to me, the cleanest solution would indeed be a double for loop over rows and columns. The only Python-level operation in that would be setting the value, everything else would be in C (if done right, I can help). Not sure if that slowdown kills it for your application.